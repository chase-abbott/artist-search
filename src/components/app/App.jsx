import React from 'react';
import Home from '../artist/Home';
import { Switch, Route } from 'react-router-dom';
import ArtistReleases from '../detail/ArtistReleases';
import RecordingsPage from '../recordings/RecordingsPage';

export default function App() {
  return (
    <Switch>
      <Route exact path="/" component={Home}/>
      <Route exact path="/artist-details/:id" component={ArtistReleases}/>
      <Route exact path="/recordings/:id" component={RecordingsPage} />
      <Route exact path="/lyrics/:artist/:title"/>
    </Switch>
  );
}
