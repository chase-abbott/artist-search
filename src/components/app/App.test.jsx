import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { MemoryRouter as Router } from 'react-router-dom';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<Router><App /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
