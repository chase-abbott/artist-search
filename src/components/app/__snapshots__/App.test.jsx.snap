// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button
    disabled=""
  >
    &lt;
  </button>
  1
  <button>
    &gt;
  </button>
  <form>
    <input
      value=""
    />
    <button
      data-testid="search"
    >
      search
    </button>
  </form>
  <ul />
</DocumentFragment>
`;
